squirrel/sqcompiler.cpp

```diff
@@ -1855,327 +1886,1729 @@ class SQCompiler
                 Error(_SC("scalar expected : integer, float, or string"));
         }
         Lex();
-        return val;
-    }
-
-    SQTable* GetScopedConstsTable()
-    {
-        assert(!_scopedconsts.empty());
-        SQObjectPtr &consts = _scopedconsts.top();
-        if (sq_type(consts) != OT_TABLE)
-            consts = SQTable::Create(_ss(_vm), 0);
-        return _table(consts);
+        return ret;
     }
 
-    void ConstStatement(bool global)
+    ConstDecl *parseConstStatement(bool global)
     {
         Lex();
-        SQObject id = Expect(TK_IDENTIFIER);
-        bool ignoreGlobalConst = global && !(_fs->lang_features & LF_FORBID_GLOBAL_CONST_REWRITE);
-        CheckDuplicateLocalIdentifier(id, _SC("Constant"), ignoreGlobalConst);
+        Id *id = (Id *)Expect(TK_IDENTIFIER);
+        bool ignoreGlobalConst = global && !(_lang_features & LF_FORBID_GLOBAL_CONST_REWRITE);
```

Comment:
Not needed since the check for duplicated was moved to CodegenVisitor::visitConstDecl()

Reply:
fixed